tests/conftest.py

```diff
@@ -25,6 +25,20 @@
 from litserve.utils import wrap_litserve_start
 
 
+def pytest_collection_modifyitems(config, items):
+    """Auto-mark tests based on their file path."""
+    for item in items:
+        # Get the relative path of the test file
+        test_path = item.fspath.relto(config.rootdir)
+
+        if "tests/unit/" in str(test_path):
+            item.add_marker(pytest.mark.unit)
+        elif "tests/integration/" in str(test_path):
+            item.add_marker(pytest.mark.integration)
+        elif "tests/e2e/" in str(test_path):
+            item.add_marker(pytest.mark.e2e)
```

Comment:
nice :) but since they are in each folder, why do you need marker and not just run the folder?

Reply:
> I am more worried about the confusion that someone will place a test marked as e2e un unit folder etc.

Not sure how will that happen when the tests are clearly separated into folders. This pytest marking is just an addon.

Reply:
If I want to run unit tests which are async, I can do  `pytest -m "unit and asyncio"`.